enhancements/shared-and-guaranteed-cpu-pools.md

```diff
@@ -0,0 +1,407 @@
+to calculate CFS shares/quotas/etc when necessary.
+
+However, there would be significant impacts to other kubernetes components to handle the new
+`cpu-guaranteed` resource in parallel to the existing `cpu` resource. This includes the apiserver,
+scheduler, pod autoscaling and more.
+
+Given that this change is being done for a very specific usecase, there is no chance the
+upstream community would accept a change of this magnitude.
+
+A variation of this alternative would be to add a new extended resource for the guaranteed CPUs 
+(e.g. `openshift.io/guaranteed-cpus`) but continue to use the existing `cpu` resource for shared 
+CPUs. This actually makes things worse as it would require just as many changes as would be
+required to add the new first class resource, but most of the changes would have to be made
+downstream and carried in patches indefinitely.
+
+### Extend Workload Partitioning
+
+The existing [Management Workload Partitioning](https://github.com/openshift/enhancements/blob/master/enhancements/workload-partitioning/management-workload-partitioning.md) 
+could be extended to add support for a new `shared` partition. To place a pod in the `shared` 
+partition, the user would annotate the namespace with `workload.openshift.io/allowed: shared` and 
+then annotate each pod with `target.workload.openshift.io/management: {"effect": "PreferredDuringScheduling"}`. 
+
+Here is the workflow:
+1. The cluster creator creates a Performance Profile for the NTO and specifies a shared CPU partition.
+2. The cluster creator creates a machine config manifest to configure CRI-O to partition shared workloads.
+3. The cluster creator creates a machine config manifest to write a configuration file for kubelet to specify
+the shared CPUs.
+4. The cluster creator then creates the cluster.
+5. The kubelet starts up, finds the configuration file and removes the shared CPUs from the `defaultCpuSet`,
+adding them to the `reservedSystemCPUs` instead. 
+6. The kubelet advertises `shared.workload.openshift.io/cores` extended resources on the node based on the 
+number of shared CPUs from the config file.
+7. Something schedules a pod with the `target.workload.openshift.io/shared` annotation in a namespace 
+with the `workload.openshift.io/allowed: shared` annotation. 
+   * The admission hook modifies the containers in the pod:
+     * replacing the `cpu` requests with `shared.workload.openshift.io/cores` requests/limits
+     * adding the `resources.workload.openshift.io/{container-name}: {"cpushares": <shares>, "cpulimits": <limits>}`
+   annotation for CRI-O.
+   * The scheduler sees the new pod and finds available `shared.workload.openshift.io/cores` resources on the node. 
+   The scheduler places the pod on the node.
+   * Kubelet does not do the cgroup configuration for the containers since it is a shared pod (based on the annotation).
+   * CRI-O does the cgroup configuration for the containers in the pod based on the shares/limits annotated 
```

Comment:
maybe a note here saying this is the same workflow as solely guaranteed pods were in the past

Reply:
Yes - I will add that.